File: type_system.js
// Filename: type_system_js.js
function combine(a, b) {
    let result = a + b;
    console.log(`JavaScript Result (${typeof a} + ${typeof b}): ${result}`);
  }
  
  combine(10, 5);          // Number addition
  combine("Hello, ", "World!"); // String concatenation
  combine(10, "World!");   // Coerces number to string and concatenates

File: type_system.py
# Filename: type_system_py.py
def combine(a, b):
    try:
        result = a + b
        print(f"Python Result ({type(a).__name__} + {type(b).__name__}): {result}")
    except TypeError as e:
        print(f"Python Error: {e}")

combine(10, 5)          # Integer addition
combine("Hello, ", "World!") # String concatenation
combine(10, "World!")   # Throws a TypeError